src/librbd/ImageCtx.cc

```diff
@@ -552,15 +555,19 @@ struct C_InvalidateCache : public Context {
     SnapInfo info(in_snap_name, in_snap_namespace,
 		  in_size, parent, protection_status, flags);
     snap_info.insert(pair<snap_t, SnapInfo>(id, info));
-    snap_ids.insert(pair<string, snap_t>(in_snap_name, id));
+    snap_ids.insert(pair<pair<cls::rbd::SnapshotNamespace, string>,
```

Comment:
Nit: good example of using C++11 brace initialization to make the code look cleaner